### ISSUE TITLE
undefined method `roles' for nil:NilClass

### ISSUE THREAD

**Issue by devaniljr** [enjaku4/rabarber#36]
Hello, I'm having a problem. I have my own authentication, and I access the current_user through a method that looks up the logged-in user via session. However, this is giving me the following error:

```
undefined method `roles' for nil:NilClass

Hint: Something is nil when it probably shouldn't be.
verify_accessrabarber (1.2.0) lib/rabarber/controllers/concerns/authorization.rb

def verify_access
  Rabarber::Missing::Actions.new(self.class).handle
  Rabarber::Missing::Roles.new(self.class).handle
  return if Rabarber::Permissions.access_granted?(
    send(Rabarber::Configuration.instance.current_user_method).roles, self.class, action_name.to_sym, self
  )
  Rabarber::Configuration.instance.when_unauthorized.call(self)
end

```

I believe the problem is because it expects the current_user method to always return a user to be able to access the "roles" relation, which doesn't happen when it doesn't find a User. What's the best strategy to get around this problem?

**Comment by enjaku4** [comment]
Hello @devaniljr, thank you for bringing this up. This issue was fixed in version 1.2.1, so updating to the latest version should help you. When the user is not authenticated (`current_user` returns `nil`), Rabarber treats such cases as if it was the user without any roles.

**Comment by devaniljr** [comment]
Great. It's working now. Thank you. I'm loving the gem
